File: GeometryCalcJava/src/amdegregorio/geo/GeometryCalc.java
package amdegregorio.geo;

public class GeometryCalc {

   public static void main(String[] args) {
      //Default values if no arguments supplied
      int length = 42;
      int width = 20;
      
      if (args.length > 0) {
         try {
            //If only one argument supplied - it's a square use it for length and width
            if (args.length == 1) {
               length = Integer.parseInt(args[0]);
               width = Integer.parseInt(args[0]);
            } else {
               length = Integer.parseInt(args[0]);
               width = Integer.parseInt(args[1]);
            }
         } catch (NumberFormatException nfe) {
            System.err.println("Usage: java GeometryCalc [length] [width]");
            System.err.println("  length: a whole number representing the length of a rectangle");
            System.err.println("  width: a whole number representing the width of a rectangle");
            System.exit(1);
         }
      }
      
      Rectangle rectangle = new Rectangle(length, width);
      System.out.println("Created rectangle: " + rectangle.toString());
      System.out.println("Area: " + rectangle.calculateArea());
      System.out.println("Perimeter: " + rectangle.calculatePerimeter());
      System.exit(0);
   }

}

File: GeometryCalcScala/src/amdegregorio/geo/GeometryCalc.scala
package amdegregorio.geo

object GeometryCalc {
  def main(args: Array[String])  {
    var length: Int = 42
    var width: Int = 20
    
    if(args.length > 0) {
      try {
          val intArgs = args.map(_.toInt)
          if (args.length == 1) {
            length = intArgs(0)
            width = intArgs(0)
          } else {
            length = intArgs(0)
            width = intArgs(1)
          }
      } catch {
        case ex: NumberFormatException => {
          println("Usage: scala GeometryCalc [length] [width]")
          println("  length: a whole number representing the length of a rectangle")
          println("  width: a whole number representing the width of a rectangle")
          System.exit(1)
        }
      }
    }
    
    val rectangle = new Rectangle(length, width)
    println("Created rectangle: " + rectangle.toString());
    println("Area: " + rectangle.calculateArea())
    println("Perimeter: " + rectangle.calculatePerimeter())
  }
}